include/utility/json_utility.hpp

```diff
@@ -926,5 +926,68 @@ inline bool isFruPowerOffOnly(const nlohmann::json& i_sysCfgJsonObj,
             (i_sysCfgJsonObj["frus"][i_vpdFruPath].at(0)["powerOffOnly"]));
 }
 
+/**
+ * @brief API which tells if the FRU 's VPD is recollectable at runtime
+ *
+ * @param[in] i_sysCfgJsonObj - System config JSON object.
+ * @param[in] i_dbusObjPath - D-bus object path.
+ *
+ * @return true if FRU 's VPD is recollectable at runtime. false otherwise.
+ */
+inline bool isVPDRecollectableAtRuntime(const nlohmann::json& i_sysCfgJsonObj,
+                                        const std::string& i_dbusObjPath)
+{
+    try
+    {
+        if (i_dbusObjPath.empty())
+        {
+            throw std::runtime_error("Given D-bus object path is empty.");
+        }
+
+        if (i_sysCfgJsonObj.empty())
+        {
+            throw std::runtime_error(
+                "System config JSON object is not present. Failed checking if the FRU VPD is recollectable at runtime or not for " +
+                std::string(i_dbusObjPath));
+        }
+
+        if (!i_sysCfgJsonObj.contains("frus"))
+        {
+            throw std::runtime_error(
+                "System config JSON is missing frus section.");
+        }
+
+        const nlohmann::json& l_listOfFrus =
+            i_sysCfgJsonObj["frus"].get_ref<const nlohmann::json::object_t&>();
+
+        for (const auto& l_frus : l_listOfFrus.items())
+        {
+            for (const auto& l_inventoryItem : l_frus.value())
+            {
+                if (i_dbusObjPath.compare(l_inventoryItem["inventoryPath"]) ==
+                    constants::STR_CMP_SUCCESS)
+                {
+                    if (!l_inventoryItem.contains("vpdRecollectableAtRuntime"))
```

Comment:
Instead  of throwing exception, should we check for "vpdRecollectableAtRuntime" tag presence , if found return its bool value.

anyway we are returning false, it applies to both incase tag not found or objectpath not found.